dashboard/lib/announcements.rb

```diff
@@ -0,0 +1,60 @@
+class Announcements
+  @@announcements_data = nil
+  @@loaded = false
+  @@load_error = false
+  @@json_path = File.join("#{pegasus_dir}/sites.v3/code.org", 'announcements.json')
+
+  # enables unit tests
+  def self.set_file_path(path)
+    @@json_path = path
+    @@loaded = false
+    @@load_error = false
+  end
+
+  # gets special announcement data for a page, or nil if not found
+  def self.get_announcement_for_page(page)
+    load_announcements
+    return nil if @@load_error || !@@announcements_data
+    pages = @@announcements_data[:pages]
+    banners = @@announcements_data[:banners]
+    page = page.to_sym
+    return nil unless pages[page]
+
+    banner = banners[pages[page].to_sym]
```

Comment:
If you want to avoid all the `to_sym` concerns, take a look at [`HashWithIndifferentAccess`](https://api.rubyonrails.org/classes/ActiveSupport/HashWithIndifferentAccess.html) or [`OpenStruct`](https://ruby-doc.org/stdlib-2.5.1/libdoc/ostruct/rdoc/OpenStruct.html).